### PR TITLE
Feature/add resource names v2

### PR DESCRIPTION
**Pull request by kevin11-allen-nttd**
I figure we should handle different versions of this module to avoid breaking changes since its used in so many modules, but let me know if you feel differently

### CODE REVIEW

File: locals.tf [hunk 1]
```diff
@@ -4,8 +4,10 @@ locals {
   }
   tags = merge(local.default_tags, var.tags)
 
-  resource_group_name  = module.resource_names["resource_group"].standard
-  virtual_network_name = module.resource_names["virtual_network"].standard
+  use_v2_resource_names = (var.resource_names_version == "2")
```
Comment (aaron-christian-nttd) [inline locals.tf:7]: Wouldn't a boolean be better here?

Comment (kevin11-allen-nttd) [reply locals.tf:7]: do you mean for the variable? I suppose that would work, but I am not sure why that would be better

Comment (kevin11-allen-nttd) [reply locals.tf:7]: in the case where we want to support resource names v3 for example, we would either need to add another boolean to the locals or add another boolean in the variables. my hunch was to keep it to one variable and handle it in the locals

Comment (ben-vaughan-nttd) [reply locals.tf:7]: This is the same thought I had. IMO it's good to leave it open for future changes.

**Review by ben-vaughan-nttd — APPROVED** [review]

**Review by aaron-christian-nttd — APPROVED** [review]

**Pull request merged**